packages/fether-react/src/Send/TxForm/TxForm.js

```diff
@@ -378,7 +380,7 @@ class TxForm extends Component {
    * Prevalidate form on user's input. These validations are sync.
    */
   preValidate = values => {
-    const { balance, token } = this.props;
+    const { balance, chainId: currentChainIdBN, token } = this.props;
```

Comment:
cc @amaurymartiny the chainId$() rpc returns a bignumber, but AFAIK we could just return a regular int and it might simplify post treatment; no need for bignumber.

what's the consensus on this? are all numbers returned from `@parity/api` BigNumber? if so we should keep it this way; if not do we want to return regular ints for "simple" numbers? or would it be more confusing?

Reply:
I think it could be a regular in too, but all the others RPC calls need to be BigNumber so it would be the only exception (i.e. to check balance, block number, transaction count, peer count, etc), so I agree with you that we should keep it as a BigNumber.

FYI, I had a look around and found that the `chainId$` comes @paritytech/ light.js in https://github.com/paritytech/js-libs/blob/master/packages/light.js/src/rpc/eth.ts#L21, where the Typescript indicates returns type `BigNumber | Symbol` (in that file it looks like other calls also return this type), which calls from rpc/eth in @parity/ api https://github.com/paritytech/js-libs/blob/master/packages/api/src/rpc/eth/eth.js#L42 in light.js.
The API docs show that this type is returned here https://paritytech.github.io/js-libs/light.js/api/modules/_rpc_eth_.html
In the Parity Ethereum Rust codebase under rpc/eth the `eth_chainId` returns `U64` https://github.com/paritytech/parity-ethereum/blob/master/rpc/src/v1/traits/eth.rs#L55

Reply:
Sorry, I just saw this.

> are all numbers returned from @parity/api BigNumber?

Yes, and I would keep it like that for consistency. Trick: instead of using `.valueOf()` `.toNumber()` or anything else, `+chainId` also works.